### PR TITLE
Acceptance tests to verify dataset modification flows

### LINKED ISSUE #1536: Acceptance tests to verify dataset modification flows
As a Renku maintainer, I'd like acceptance tests to verify the new dataset modification flows which create new instance of a dataset when it's changed.

**Acceptance criteria:**
- [ ] make sure there are acceptance-tests verifying the dataset modification flow:
  * create a project with a dataset, modify the dataset and modify the modification; as a result, only the latest modification should be displayed on the project's datasets screen.

### PR DESCRIPTION
**Pull request by joke1196**
closes #1536

### CODE REVIEW

File: acceptance-tests/src/test/scala/ch/renku/acceptancetests/DatasetModificationFlow.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import ch.renku.acceptancetests.generators.Generators.paragraph
+import ch.renku.acceptancetests.model.datasets.{DatasetName, DatasetTitle}
+import ch.renku.acceptancetests.model.projects.ProjectDetails
+import ch.renku.acceptancetests.pages.ProjectPage
+import ch.renku.acceptancetests.tooling.{AcceptanceSpec, DocsScreenshots}
+import ch.renku.acceptancetests.workflows._
+
+import scala.language.postfixOps
+
+class DatasetModificationFlow extends AcceptanceSpec with Login with NewProject with RemoveProject with Datasets {
```
Comment (jachro) [inline acceptance-tests/src/test/scala/ch/renku/acceptancetests/DatasetModificationFlow.scala:13]: Should we maybe have the `Spec` suffix in the name of the class? I think all the current specs have it and not sure if sbt will pick it up when it's not there...

File: acceptance-tests/src/test/scala/ch/renku/acceptancetests/DatasetModificationFlow.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  scenario("From the UI the user can modify a dataset and only interact with its latest version") {
+    import Modification._
+    implicit val loginType:       LoginType       = logIntoRenku
+    implicit val docsScreenshots: DocsScreenshots = new DocsScreenshots(this, browser)
+    implicit val projectDetails:  ProjectDetails  = ProjectDetails.generate()
+
+    implicit val projectPage = ProjectPage()
+    val datasetName          = DatasetName.generate
+    val newDescription       = paragraph().generateOne
+    createNewProject(projectDetails)
```
Comment (jachro) [inline acceptance-tests/src/test/scala/ch/renku/acceptancetests/DatasetModificationFlow.scala:24]: Maybe `Given("a renku project exists")` or smth better ;)

File: acceptance-tests/src/test/scala/ch/renku/acceptancetests/DatasetModificationFlow.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    Given("a new dataset")
+    val originalDatasetPage = `create a dataset`(datasetName)
+
+    val newTitle = DatasetTitle.generate
+    When("the user modifies the dataset")
+    `modify the dataset`(originalDatasetPage,
+                         by = `changing its title`(to = newTitle.toString),
```
Comment (jachro) [inline acceptance-tests/src/test/scala/ch/renku/acceptancetests/DatasetModificationFlow.scala:32]: Wow! I like this DSL for ds modification!

File: acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala [hunk 4]
```diff
@@ -19,8 +25,90 @@ trait Datasets {
...
+    And("the user saves the modification")
+    click on newDatasetPage.ModificationForm.datasetSubmitButton sleep (10 seconds)
+
+    val datasetPage = DatasetPage(datasetName)
+    Then("the user should see its newly created dataset and the project it belongs to")
+    verify browserAt datasetPage
+    verify that datasetPage.datasetTitle contains datasetName.value
+    val projectLink = datasetPage.ProjectsList.link(projectPage)(_: WebDriver)
+    reload whenUserCannotSee projectLink
+    verify userCanSee datasetPage.ProjectsList.link(projectPage)
```
Comment (jachro) [inline acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala:56]: Do we need to verify the link presence again after the reloading passes?

File: acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala [hunk 5]
```diff
@@ -19,8 +25,90 @@ trait Datasets {
...
+  ): DatasetPage = {
+    Given(s"the user is on the page of the dataset")
+    `navigate to dataset`(datasetPage)
+
+    When(s"the user clicks on the modify button")
+    click on datasetPage.modifyButton
+    verify userCanSee datasetPage.ModificationForm.formTitle
+
+    And(s"the user modifies the dataset by ${by.name}")
+    by.execute(datasetPage)
```
Comment (jachro) [inline acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala:84]: Could we make this piece where we modify a bit more readable? Not so sure if I understand what's going on here.

File: acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala [hunk 6]
```diff
@@ -19,8 +25,90 @@ trait Datasets {
...
+    }
+
+    And("the user saves the modification")
+    click on datasetPage.ModificationForm.datasetSubmitButton
+
+    Then("the user should see its dataset and to which project it belongs")
+    verify browserAt datasetPage
+    val projectLink = datasetPage.ProjectsList.link(projectPage)(_: WebDriver)
+    reload whenUserCannotSee projectLink
+    verify userCanSee datasetPage.ProjectsList.link(projectPage)
```
Comment (jachro) [inline acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala:97]: The same as in some comment above. Do we need to verify if the link is visible after the reloading succeeded?

File: acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala (hunk 4 above)
Comment (joke1196) [reply acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala:56]: Yes I don't think we do. I will also remove it from verifyDatasetCreated fonction

File: acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala (hunk 5 above)
Comment (joke1196) [reply acceptance-tests/src/test/scala/ch/renku/acceptancetests/workflows/Datasets.scala:84]: Agreed. I think I'll go for: `And(s"the user modifies the dataset by ${by.name}")
    by.modifying(datasetPage)`

**Review by jachro — APPROVED** [review]
Cool! Great job!

**Pull request merged**